src/IO/SensorLogger.cpp

```diff
@@ -42,25 +42,47 @@ void SensorLogger::run()
         boost::filesystem::create_directories(path);
     }
 
-    // bool shouldWriteHeader = !std::experimental::filesystem::exists(path +
-    // filename); fileStream = std::make_shared<std::ofstream>(path + filename,
-    // std::ios_base::app);
-
-    // if (shouldWriteHeader)
-    // {
-    // 	std::ofstream fileStream;
-    // 	fileStream.open(path + filename, std::ios_base::app);
-    // 	writeHeader(fileStream);
-    // 	fileStream.close();
-    // }
+    std::string filename = std::to_string(bootId) + ext;
+    bool shouldWriteHeader = !boost::filesystem::exists(path + filename);
+    std::ofstream fileStream{path + std::to_string(bootId) + ext, std::ios_base::ate};
+
+    /*if (shouldWriteHeader)
+    {
+        std::ofstream fileStream;
+        fileStream.open(path + filename, std::ios_base::app);
+        writeHeader(fileStream);
+        fileStream.close();
+    }*/
 
     status.fileStatus = READY;
-    std::ofstream fileStream{path + std::to_string(bootId) + ext, std::ios_base::ate};
+
+    bool isFirstLine = true;
+
     while (true)
     {
         if (!logQueue.empty())
         {
-            dequeueToFile(fileStream);
+
+            // get data
+            sensorsData currentState = getCurrentState();
+
+            // write the headers if necessary
+            if (shouldWriteHeader && isFirstLine)
+            {
+                writeToHeader(currentState, isFirstLine, path, filename);
```

Comment:
`isFirstLine` should not be passed to the function. It should be updated here in the loop.